lib/matplotlib/testing/decorators.py

```diff
@@ -504,3 +504,62 @@ def _image_directories(func):
     result_dir = Path().resolve() / "result_images" / module_path.stem
     result_dir.mkdir(parents=True, exist_ok=True)
     return baseline_dir, result_dir
+
+
+def generate_matplotlib_baseline_images_from_previous_commit():
+    mplRepoPath = Path(mpl.__file__).parent
+    # Check that git is installed.
+    subprocess.run(shutil.which('git'),
+                    stderr=subprocess.STDOUT,
+                    stdout = open(os.devnull, 'w'),
+                    universal_newlines=True) == None
+    # Check if current repository is a git repository
+    subprocess.run(['git', 'status'],
+                    cwd = mplRepoPath,
+                    stderr=subprocess.STDOUT,
+                    stdout = open(os.devnull, 'w'),
+                    universal_newlines=True) == None
+    # Create virtual env in tmpdir.
+#    if not (mplRepoPath / 'tmp/testenv').exists():
+#        modify_existing_baseline_images()
+#    else
+        subprocess.run(["python", "-mvenv", "tmp/testenv"],
```

Comment:
Don't rely on `'python'` being in the path or being the _right_ Python, it is better to use `sys.executable`